cargo-shuttle/src/lib.rs

```diff
@@ -486,203 +684,173 @@ impl Shuttle {
+                    ?stop_request,
+                    "stopping service because it received a signal"
+                );
+                let _response =
+                    rt.1.stop(tonic::Request::new(stop_request))
+                        .or_else(|err| async {
+                            provisioner_server.abort();
+                            rt.0.kill().await?;
+                            Err(err)
+                        })
+                        .await?
+                        .into_inner();
             }
 
-            let resources = response
-                .resources
-                .into_iter()
-                .map(resource::Response::from_bytes)
-                .collect();
-
-            println!("{}", get_resources_table(&resources, service_name.as_str()));
-
-            let mut stream = runtime_client
-                .subscribe_logs(tonic::Request::new(SubscribeLogsRequest {}))
-                .or_else(|err| async {
-                    provisioner_server.abort();
-                    runtime.kill().await?;
-
-                    Err(err)
-                })
-                .await?
-                .into_inner();
-
-            tokio::spawn(async move {
-                while let Ok(Some(log)) = stream.message().await {
-                    let log: shuttle_common::LogItem = log.try_into().expect("to convert log");
-                    println!("{log}");
-                }
-            });
+            return Ok(());
+        }
 
-            let addr = SocketAddr::new(
-                if run_args.external {
-                    Ipv4Addr::new(0, 0, 0, 0)
-                } else {
-                    Ipv4Addr::LOCALHOST
+        // If no signal was received during runtimes initialization, then we must handle each runtime until
+        // completion and handle the signals during this time.
+        if cfg!(target_family = "unix") {
+            let sigterm = sigterm_notif.as_mut().expect("SIGTERM reactor failure");
+            let sigint = sigint_notif.as_mut().expect("SIGINT reactor failure");
+            for mut rt in runtimes {
+                // If we received a signal while waiting for any runtime we must stop the rest and exit
+                // the waiting loop.
+                if signal_received {
+                    let stop_request = StopRequest {};
+                    trace!(
+                        ?stop_request,
+                        "stopping service because it received a signal"
+                    );
+                    let _response =
+                        rt.1.stop(tonic::Request::new(stop_request))
+                            .or_else(|err| async {
+                                provisioner_server.abort();
+                                rt.0.kill().await?;
+                                Err(err)
+                            })
+                            .await?
+                            .into_inner();
+                    continue;
                 }
```

Comment:
The above handling refers to receiving a signal during the runtimes initializations. If `signal_received` is `true` it should exit the `local_run` returning `Ok(())`. Otherwise, it passes it and moves to the logic of waiting on each `runtime` in a for loop. However, while waiting on these runtimes we must handle signals as well, hence we're checking the `signal_received` before actually waiting on a runtime in a new iteration loop. If it's true, then we've received a signal in a previous iteration and we must stop all the runtimes which we haven't waited on instead on continuing to wait on them.